Add new Tests and update Docs

ddspog opened the pull request:
# Proposed Changes

I've added tests for components Auth, Login, PartyRsvp, PartyUninvited, Password and Register, closing issue #6.

I've updated contributing file, with templates and emojis, and removing labels section, closing issue #10. 

I've updated issue and pull request templates, with more information on some sections, closing issue #12.
### Type of changes
- Review Docs (better documentation)
- Testing (automatic tests)
### Features Working

The tests run fast, and all pass. Writing them, allow me to fix errors on some components. 

And also allow me to introduce some code free from testing, making them unreachable during tests.
This was needed because some code only makes logs from console, or makes some states changes. Behavior that wasn't desired for testing, since it overflow the logs.
### Tests

The main tests I made, were to verify initialization of some components variables, and to check call to other methods already tested. They all run and pass from the Mocha Browser tester.
### Docs

I've updated the contributing file, adding some more emojis to put on commit. Added a commit template section. Remove the labels section, and reference the pull request template.

I've also updated the issue templates, guiding to user stories production, and epics goals.

I've also updated the pull request template, removing checkboxes.
### Reviewers:

@ddspog

ddspog committed:
Add tests to Auth component :white_check_mark:

I've added tests to Auth component, testing helpers and functions used,
with a package for testing user login.

ddspog committed:
Add package to test logins :arrow_up:

I've added accounts-testing package to make testing logins easier. This
package allows to login without password nor registrationg with a nice
method insecureUserLogin()

ddspog committed:
Merge branch 'dennis' of https://github.com/ddspog/socially into dennis

ddspog committed:
Add tests for Login component :white_check_mark:

I've added tests for Login component, using normal verification tests
that see if Meteor was called on correct places.

ddspog committed:
Rem call to stubs on Login tests :fire:

I've removed the call to stubs, that weren't necessary on Login test.

ddspog committed:
Mod all test to be sync :white_check_mark:

I've modify all tests to be sync, and them, not trying to execute all
at the same time. Thus breaking some dependencies.

ddspog committed:
Mod func to suppress error on Dev :shirt:

I've modified this configuration function, to show log on errors only
at production.

ddspog committed:
Mod submit func to fix problems :bug:

I've modified some calls, to improve code and fixing some errors.

ddspog committed:
Mod comp to show errors on Production :shirt:

I've specified that the log and erros functions will be called only on
production at this component.

ddspog committed:
Add callback to logout :bug:

I've added a callback to logout, allowing to use the callback, to run
code exactly after logout is made.

ddspog committed:
Add flag avoiding code on tests :shirt:

I've used a environment variable named TESTING to flag some code, that
I wouldn't want to run when testing. An example it's logging and error
messages that can't be stubbed.

ddspog committed:
Ref tests to avoid login bugs :bug:

I've changed the tests to make blocks ensures it logins when needed,
and logout when needed. Also I've rearranged some lines to improve
performance.

ddspog committed:
Add test to PartyRsvp :white_check_mark:

I've added test tot PartyRsvp component, testing calls on some methods.

ddspog committed:
Rem call to isTesting :fire:

I've removed this method, since it won't be used anymore.

ddspog committed:
Mod issue template making US easier :memo:

I've modified the issue template to support enhancements and user
stories, also to make it easy turn into Epics.

ddspog committed:
Mod templates for issues and PRs :memo:

I've updated the templates, for better design, and also to include
other types of issues: goal, user story or enhancement.

Fix #12

ddspog committed:
Mod contributing making corrections :memo:

I've updated the contributing file, to update some links, and making
corrections on some parts. Also the design improved. I've removed the
list of labels, since I had changed them a lot previously. Also, I've
added more orientation on commit templates, and added more emoji
options.

Fix #10

ddspog committed:
Mod PartyUninvited to don't show log on test :mute:

I've modified this component, to avoid unnecessary logging on testing.

Ref #6

ddspog committed:
Mod method to consider null users :shirt:

I've modified getContactEmail() method, allowing to verify if user
given it's null, then it it is, return null. This will help a lot on
testing purposes.

Ref #6

ddspog committed:
Mod test to ensures behavior :shirt:

I've modified this test, to make another try on completing avoiding
errors on test scope.

Ref #6

ddspog committed:
Add PartyUninvited test :white_check_mark:

I've added PartyUninvited test, checking call to some methods.

Ref #6

ddspog committed:
Add Password tests :white_check_mark:

I've added Password test, checking call to some method.

Ref #6

ddspog committed:
Mod Register supressing callback on test :shirt:

I've modified Register, a callback, putting a condition to supress
behavior on testing.

Ref #6

ddspog committed:
Mod Password to supress behavior on test :shirt:

I've modified Password to supress a callback function during tests.

Ref #6

ddspog committed:
Add Register tests :white_check_mark:

I've added tests for Register component, checking call to some methods.

Fix #6

review-ninja commented:
<a href="https://app.review.ninja/ddspog/mam-seed/pull/13" target="_blank"><img src="https://app.review.ninja/59931460/pull/13/badge" alt="ReviewNinja"/>

ddspog committed:
Merge branch 'master' into dennis

ddspog committed:
Mod DB startup with better examples :non-potable_water:

I've modified and improve DB examples. The DB now have: loaded users;
parties with location, owner, rsvps defined;

This will help improve testing.

Ref #13

ddspog committed:
Add module to get callbacks :bulb:

I've added this new module, allowing to some classes import callbacks,
for refactoring purposes. And for easy correction of future mistakes.

Ref #13

ddspog committed:
Mod classes to use AfterLogInout :art:

On all classes that uses callback, redirecting user to parties section,
I put an call to this function, and import it.

Ref #13

ddspog committed:
Mod html to ensures block to appear :lipstick:

Since I modified some of the structure of HTML, I've got into a problem
of indexing the elements. Putting the top element on center wasn't
enough, I needed to ensure some elements to appear at front.

Ref #13

ddspog committed:
Fix alignment on RSVP buttons :lipstick:

I've changed the aligment of RSVP buttons, for better display.

Ref #13

ddspog committed:
Mod spacing and identation on class :art:

I've update the spacing and indentation, using a beautify command.

Ref #13

ddspog committed:
Fix buggy layout on PartiesList :lipstick:

Since last updates on tests, I've got to see some buggy layout on the
parties liest section. So I've spend some time fixing.

Ref #13

ddspog committed:
Merge branch 'dennis' of https://github.com/ddspog/socially into dennis

ddspog committed:
Add new callback for logging :bulb:

I've added a new callback to log some msgs after some action. If the
error msg isn't defined, it prints the reason of error thrown. If the
done msg isn't defined, it prints 'Done!'.

Ref #13

ddspog committed:
Mod PartyRsvp to use callback :art:

I've modified PartyRsvp to use the callback function I've created.

Ref #13

ddspog commented on imports/ui/components/password/client/password.tests.js:
```diff
@@ -0,0 +1,72 @@
+
+            afterEach(function(done) {
+                if (spies.forgot)
+                    spies.forgot.restore();
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected '{' and instead saw 'spies'.](https://www.codacy.com/app/ddspog/MAM-Seed/file/2729684822/issues/source?bid=3340041&fileBranchId=3346427#l55)

ddspog commented on imports/ui/components/partiesList/client/partiesList.tests.js:
```diff
@@ -5,55 +5,83 @@ import {
 describe('PartiesList', function() {
-    beforeEach(function() {
+    if (!process.env.TESTING)
+        process.env.TESTING = 1;
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected '{' and instead saw 'process'.](https://www.codacy.com/app/ddspog/MAM-Seed/file/2729684801/issues/source?bid=3340041&fileBranchId=3346427#l23)

ddspog commented on imports/ui/components/partiesList/client/partiesList.tests.js:
```diff
@@ -5,55 +5,83 @@ import {
+
+    beforeEach(function(done) {
+        if (!stubs.config)
+            stubs.create('config', ConfigGoogleMap, 'apply').returns('');
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected '{' and instead saw 'stubs'.](https://www.codacy.com/app/ddspog/MAM-Seed/file/2729684801/issues/source?bid=3340041&fileBranchId=3346427#l27)

ddspog commented on imports/ui/components/login/client/login.tests.js:
```diff
@@ -0,0 +1,94 @@
+
+            afterEach(function(done) {
+                if (spies.login)
+                    spies.login.restore();
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected '{' and instead saw 'spies'.](https://www.codacy.com/app/ddspog/MAM-Seed/file/2729684798/issues/source?bid=3340041&fileBranchId=3346427#l54)

ddspog commented on imports/ui/components/register/client/register.tests.js:
```diff
@@ -0,0 +1,100 @@
+
+            afterEach(function(done) {
+                if (spies.register)
+                    spies.register.restore();
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected '{' and instead saw 'spies'.](https://www.codacy.com/app/ddspog/MAM-Seed/file/2729684824/issues/source?bid=3340041&fileBranchId=3346427#l60)

ddspog commented on imports/ui/components/register/client/register.tests.js:
```diff
@@ -0,0 +1,100 @@
+    spies.restoreAll();
+
+    if (!process.env.TESTING)
+        process.env.TESTING = 1;
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected '{' and instead saw 'process'.](https://www.codacy.com/app/ddspog/MAM-Seed/file/2729684824/issues/source?bid=3340041&fileBranchId=3346427#l22)

ddspog commented on imports/ui/components/partyAdd/client/partyAdd.tests.js:
```diff
@@ -21,75 +25,114 @@ import {
+
+            afterEach(function(done) {
+                if (spies.insert)
+                    spies.insert.restore();
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected '{' and instead saw 'spies'.](https://www.codacy.com/app/ddspog/MAM-Seed/file/2729684806/issues/source?bid=3340041&fileBranchId=3346427#l116)

ddspog commented on imports/ui/components/partyAdd/client/partyAdd.tests.js:
```diff
@@ -21,75 +25,114 @@ import {
+                if (spies.insert)
+                    spies.insert.restore();
+                if (spies.reset)
+                    spies.reset.restore();
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected '{' and instead saw 'spies'.](https://www.codacy.com/app/ddspog/MAM-Seed/file/2729684806/issues/source?bid=3340041&fileBranchId=3346427#l118)

ddspog commented on imports/ui/components/partyAdd/client/partyAdd.tests.js:
```diff
@@ -21,75 +25,114 @@ import {
+        password: 'userCreatedPassword',
+        create: false,
+        _id: ''
+    }
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Missing semicolon.](https://www.codacy.com/app/ddspog/MAM-Seed/file/2729684806/issues/source?bid=3340041&fileBranchId=3346427#l33)

ddspog commented on imports/ui/components/partyRsvp/client/partyRsvp.tests.js:
```diff
@@ -0,0 +1,102 @@
+        password: 'userCreatedPassword',
+        create: false,
+        _id: ''
+    }
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Missing semicolon.](https://www.codacy.com/app/ddspog/MAM-Seed/file/2729684816/issues/source?bid=3340041&fileBranchId=3346427#l24)
